server/data/data-library/src/main/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTable.java

```diff
@@ -215,40 +212,40 @@ public void addAddressMapping(String user, String domain, String address) throws
             throw new RecipientRewriteTableException("Invalid emailAddress: " + address, e);
         }
         checkMapping(user, domain, MappingImpl.address(address));
-        LOGGER.info("Add address mapping => {} for user: {} domain: {}", address, user, domain);
+        LOGGER.info("Add address mapping => {} for user: {} domain: {}", address, user, domain.name());
         addMappingInternal(user, domain, MappingImpl.address(address));
 
     }
 
     @Override
-    public void removeAddressMapping(String user, String domain, String address) throws RecipientRewriteTableException {
+    public void removeAddressMapping(String user, Domain domain, String address) throws RecipientRewriteTableException {
         if (address.indexOf('@') < 0) {
             try {
-                address = address + "@" + domainList.getDefaultDomain();
+                address = address + "@" + domainList.getDefaultDomain().asString();
```

Comment:
Maybe we can have a **toMailAddress** method and relying on strong typing here (separated commit?)

Reply:
I'm not sure we can do that because sometime we need domain normalization, sometimes not.

Reply:
Ok, I agree this proposition was a bit too much. Let's get this interesting changeset merged first. We can still enhance it later if we think it makes sense.